pkg/isoeditor/rhcos.go

```diff
@@ -129,7 +169,7 @@ func fixGrubConfig(rootFSURL, extractDir string) error {
 	}
 
 	// Edit config to add custom ramdisk image to initrd
-	if err := editFile(foundGrubPath, `(?m)^(\s+initrd) (.+| )+$`, fmt.Sprintf("$1 $2 %s", ramDiskImagePath)); err != nil {
+	if err := editFile(foundGrubPath, `(?m)^(\s+initrd) (.+| )+$`, fmt.Sprintf("$1 $2 %s %s", ramDiskImagePath, nmstateDiskImagePath)); err != nil {
```

Comment:
I think it should be added only if the nmstate disk image was generated. Boot would probably fail with a missing ram disk file.